torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -85,6 +85,11 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   if (tensor.type().backend() != Backend::CPU) {
     throw TypeError("NumPy conversion for %s is not supported", tensor.type().toString().c_str());
   }
+  if (tensor.requires_grad()) {
+    throw std::runtime_error(
+        "Can't call numpy() on Variable that requires grad. "
```

Comment:
This error message is outdated. There are no more `Variables` in PyTorch.

Reply:
There are in C++ still, see https://github.com/pytorch/pytorch/issues/13638.  We should fix these once that proposal is implemented, though.